effekt/shared/src/main/scala/effekt/core/optimizer/RemoveTailResumptions.scala

```diff
@@ -35,7 +35,9 @@ object RemoveTailResumptions {
       } && default.forall { stmt => tailResumptive(k, stmt) }
       case Stmt.Region(BlockLit(tparams, cparams, vparams, bparams, body)) => tailResumptive(k, body)
       case Stmt.Alloc(id, init, region, body) => tailResumptive(k, body) && !freeInExpr(init)
-      case Stmt.Var(ref, init, capture, body) => tailResumptive(k, body) && !freeInExpr(init)
+      // Conceptually, a mutable variable definition can be seen as a handler for get and put operations.
+      // Treating this as tail-resumptive leads to a failure of semantics preservation
+      case Stmt.Var(ref, init, capture, body) => false
```

Comment:
Not too relevant for the current PR, but would a use-mention distinction help here?
If we have variables/prompts that are merely mentioned, can we still consider the function to be tail-res?

As a very trivial example, is it safe to do something similar to the example, but not use `x` at all after declaring it? 🤔